extensions/azure/processors/DeleteAzureDataLakeStorage.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file DeleteAzureDataLakeStorage.cpp
+ * DeleteAzureDataLakeStorage class implementation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "DeleteAzureDataLakeStorage.h"
+
+#include "utils/ProcessorConfigUtils.h"
+#include "utils/gsl.h"
+#include "core/Resource.h"
+
+namespace org::apache::nifi::minifi::azure::processors {
+
+const core::Relationship DeleteAzureDataLakeStorage::Success("success", "If file deletion from Azure storage succeeds the flowfile is transferred to this relationship");
+const core::Relationship DeleteAzureDataLakeStorage::Failure("failure", "If file deletion from Azure storage fails the flowfile is transferred to this relationship");
+
+void DeleteAzureDataLakeStorage::initialize() {
+  // Set the supported properties
+  setSupportedProperties({
```

Comment:
I think it might be more readable if we initialize the Properties in the class that it is defined in
e.g. setSupportedProperties(AzureStorageCredentialsService) in AzureStorageProcessorBase::initialize and in the derived class call the base::initialize and call updateSupportedProperties with the derived specific properties

Reply:
There was a previous discussion about this with @szaszm in the PR comment https://github.com/apache/nifi-minifi-cpp/pull/1158#discussion_r710018162. As there were different solutions in different processors (initializing in the base constructor, initializing in the base initialize() function then updating the properties, or initializing only in the inherited processor) we decided to go with setting all of the properties in the derived class because it would be more explicit to see all the supported properties of the processor in one place.